profiler.sh

```diff
@@ -6,10 +6,11 @@ usage() {
     echo "  start             start profiling and return immediately"
     echo "  stop              stop profiling"
     echo "  status            print profiling status"
+    echo "  list              list available profiling events"
     echo "  collect           collect profile for the specified period of time"
     echo "                    and then stop (default action)"
     echo "Options:"
-    echo "  -m mode           profiling mode: cpu|heap"
+    echo "  -e event          profiling event: cpu|alloc|lock|cache-misses etc."
```

Comment:
I agree that generally we should be careful about changing API. This feature is planned for the release 1.1, and it will be documented in change log that command-line parameters are changed.
Since we've changed 'modes' to 'events', I believe 'alloc' is a better name. 'heap' is obviously not an event.